diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SingleLineStatementsRule.kt

```diff
@@ -44,8 +47,13 @@ class SingleLineStatementsRule : Rule("statement") {
                         if (it.treeParent.elementType == ENUM_ENTRY){
                             node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeNext)
                         } else {
-                            if (!it.isBeginByNewline())
-                                node.addChild(PsiWhiteSpaceImpl("\n"), it)
+                            if (!it.isBeginByNewline()) {
+                                val nextNode = it.treeNext ?: it.parent({ parent -> parent.treeNext != null})
+                                if ( nextNode != null && nextNode.elementType == WHITE_SPACE)
+                                    (nextNode as LeafPsiElement).replaceWithText("\n")
```

Comment:
```suggestion
                                    (nextNode as LeafPsiElement).replaceWithText("\n${nextNode.text}")
```
The next white space can have spaces or newlines in it, we want to merge with them and not to replace.